client/components/tinymce/plugins/wpcom-view/views/simple-payments/index.jsx

```diff
@@ -19,18 +19,30 @@ import getSimplePayments from 'state/selectors/get-simple-payments';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import formatCurrency from 'lib/format-currency';
 import QuerySimplePayments from 'components/data/query-simple-payments';
+import QuerySitePlans from 'components/data/query-site-plans';
 import QueryMedia from 'components/data/query-media';
-import { hasFeature } from 'state/sites/plans/selectors';
+import { getCurrentPlan, hasFeature } from 'state/sites/plans/selectors';
 import { FEATURE_SIMPLE_PAYMENTS } from 'lib/plans/constants';
 
 class SimplePaymentsView extends Component {
 	render() {
-		const { translate, productId, product, siteId, planHasSimplePaymentsFeature } = this.props;
+		const {
+			translate,
+			productId,
+			product,
+			siteId,
+			sitePlan,
+			planHasSimplePaymentsFeature,
+		} = this.props;
 
 		if ( ! product ) {
 			return <QuerySimplePayments siteId={ siteId } productId={ productId } />;
 		}
 
+		if ( ! sitePlan ) {
```

Comment:
No need for the check here. When rendering, we always need plan data. We can easily get into a situation where in memory data, or persisted browser data is out of sync. So if we declare a QueryComponent, generally we shouldn't need any conditional. 

(QuerySimplePayments for example should probably be refactored a bit to provide an `all` boolean so we don't need the product check).

Reply:
I did use a conditional here because otherwise the UI will either flash the admin warning or the product information while loading the site's plan:

![aug-01-2018 18-32-57](https://user-images.githubusercontent.com/233601/43550255-5b0d0bda-95b9-11e8-84f9-bbcb2a10d694.gif)

I think that given how often Plan data is updated, this is a good first fix until we update `QuerySimplePayments` and the plans selectors to account for a loading state we can use to prevent this.

Reply:
Is possible to render a placeholder when plans is `null`?